packages/react-core/src/components/FileUpload/FileUpload.tsx

```diff
@@ -113,41 +131,60 @@ export const FileUpload: React.FunctionComponent<FileUploadProps> = ({
     dropzoneProps.onDropAccepted && dropzoneProps.onDropAccepted(acceptedFiles, event);
   };
 
-  const onDropRejected: DropFileEventHandler = (rejectedFiles: File[], event: React.DragEvent<HTMLElement>) => {
+  const onDropRejected = (rejectedFiles: FileRejection[], event: DropEvent) => {
     if (rejectedFiles.length > 0) {
-      onChange('', rejectedFiles[0].name, event);
+      onChange('', rejectedFiles[0].file.name, event);
     }
     dropzoneProps.onDropRejected && dropzoneProps.onDropRejected(rejectedFiles, event);
   };
 
+  let textInput: any;
```

Comment:
Instead of using this value that defaults to `ref: ''` (which relies on React's legacy ref system) can you use `const inputRef = inputProps.ref || React.useRef<HTMLInputElement>();` instead?